rust/routee-compass-core/src/model/label/label_model.rs

```diff
@@ -0,0 +1,47 @@
+use std::fmt::Display;
+
+use allocative::Allocative;
+use serde::Serialize;
+
+use crate::model::{
+    label::label_model_error::LabelModelError,
+    network::VertexId,
+    state::{StateModel, StateVariable},
+};
+
+#[derive(PartialEq, Eq, Hash, Debug, Clone, Serialize, Allocative)]
+pub enum Label {
+    Vertex(VertexId),
+    VertexWithIntState {
+        vertex_id: VertexId,
+        state: Vec<u64>,
```

Comment:
Was just thinking maybe we should also have a variant if we know we're only using one int to cut down on having to allocate a vec for a single number 

```rust
VertexWithSingleIntState {
  vertex_id: VertexId
  state: usize
}
```

Reply:
Sure, a few specialized instances like that could be smart since this is such a ubiquitous value type, it would certainly make a difference.